src/quasi-newton.jl

```diff
@@ -28,6 +42,32 @@ function LSR1Model(nlp::AbstractNLPModel{T, S}; kwargs...) where {T, S}
   return LSR1Model{T, S, typeof(nlp), typeof(nlp.meta)}(nlp.meta, nlp, op)
 end
 
+"""
+    DiagonalQNModel(nlp; d0 = fill!(S(undef, nlp.meta.nvar), 1.0), psb = false)
+
+Construct a `DiagonalQNModel` from another type of nlp.
+`d0` is the initial approximation of the diagonal of the Hessian.
+`psb = true` should be used to perform a PSB update (default: Andrei update). 
```

Comment:
Could you write explicitly what PSB is or link to the operator definition in LinearOperators.jl?